kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingUtils.java

```diff
@@ -532,13 +532,14 @@ private static Predicate rewriteEqualNullSafe(Column leftCol, Literal rightLit)
         "AND", new Predicate("IS_NOT_NULL", leftCol), new Predicate("=", leftCol, rightLit));
   }
 
-  /** Helper method for building DataSkippingPredicate for NOT =/<=> */
+  /** Helper method for building DataSkippingPredicate for NOT =/IS NOT DISTINCT FROM */
   private static Optional<DataSkippingPredicate> constructDataSkippingFiltersForNotEqual(
       Predicate equalPredicate,
       StatsSchemaHelper schemaHelper,
       BiFunction<Column, Literal, Optional<DataSkippingPredicate>> buildDataSkippingPredicateFunc) {
-    if (!"=".equals(equalPredicate.getName()) && !"<=>".equals(equalPredicate.getName())) {
-      throw new IllegalArgumentException("Expects predicate to be = or <=>");
+    if (!"=".equals(equalPredicate.getName())
+        && !"IS NOT DISTINCT FROM".equals(equalPredicate.getName())) {
```

Comment:
Actually here you can use `checkArgument` in `io.delta.kernel.internal.util.Preconditions` :)

Reply:
Thanks for the pointer, updated